File: ruby/rbsearch/searchoption.rb
# frozen_string_literal: true

# SearchOption - encapsulates a CLI search option
class SearchOption
  attr_reader :shortarg
  attr_reader :longarg
  attr_reader :desc
  attr_reader :func

  def initialize(shortarg, longarg, desc, func)
    @shortarg = shortarg
    @longarg = longarg
    @desc = desc
    @func = func
  end

  def sortarg
    if !@shortarg.nil? && !@shortarg.empty?
      @shortarg.downcase + 'a' + @longarg.downcase
    else
      @longarg.downcase
    end
  end

end

File: php/phpsearch/src/phpsearch/SearchOption.php
<?php declare(strict_types=1);

namespace phpsearch;

/**
 * Class SearchOption
 */
class SearchOption
{
    public string $short_arg;
    public string $long_arg;
    public string $desc;
    public ?object $func;
    public string $sort_arg;

    public function __construct(string $short_arg, string $long_arg, string $desc, $func)
    {
        $this->short_arg = $short_arg;
        $this->long_arg = $long_arg;
        $this->desc = $desc;
        $this->func = $func;
        $this->sort_arg = $this->__sort_arg();
    }

    private function __sort_arg()
    {
        if ($this->short_arg) {
            return strtolower($this->short_arg) . 'a' . $this->long_arg;
        }
        return $this->long_arg;
    }
}
